notebook/static-src/notebook/js/main.js

```diff
@@ -160,9 +129,17 @@ require([
     IPython.save_widget = save_widget;
     IPython.tooltip = notebook.tooltip;
 
+    // BEGIN HARDCODED WIDGETS HACK
+    utils.load_extension('widgets/notebook/js/extension').catch(function (err) {
+        console.warn('ipywidgets package not loaded.  Widgets are not available.  (Maybe they aren\'t installed)', err);
+    });
```

Comment:
Ah yes, hence why every other library will output HTML for them to load stuff as part of their display data.